lib/lever/client.rb

```diff
@@ -15,27 +30,48 @@ def initialize(token, options = {})
       else
         @base_uri = 'https://api.lever.co/v1'
       end
-      
+
       @options = { basic_auth: { username: token } }
 
       if options[:headers]
         @options[:headers] = options[:headers]
       end
     end
-    
+
     def users(id: nil, on_error: nil)
       get_resource('/users', Lever::User, id, { on_error: on_error })
     end
 
-    def opportunities(id: nil, contact_id: nil, on_error: nil)
+    def opportunities(id: nil, contact_id: nil, on_error: nil, hydrate_for_collection: false, **query_params)
+      # Here we're taking the first step in a larger journey to allow methods like this to return a `ResourceCollection`
+      #
+      # To start, we aim not to change current expected usage. The scenarios are:
+      # client.opportunities                            # returns an Array of Lever::Opportunity objects (unchanged)
+      # client.opportunities(contact_id: 123)           # returns an Array of Lever::Opportunity objects (unchanged)
+      # client.opportunities(id: 456)                   # returns a Lever::Opportunity (unchanged)
+      # client.opportunities(id: 456, contact_id: 123)  # returns a Lever::Opportunity (unchanged)
+      # client.opportunities(some: :param_val)          # returns a Lever::OpportunityCollection (this is new)
+      # client.opportunities(contact_id: 123, some: :param_val)           # contact_id will be added to query_params
+      # client.opportunities(id: 456, on_error: [proc], some: :param_val) # raises an error (mixing old/new interfaces)
+      #
+      if query_params.any?
+        if [id, on_error].compact.any?
+          raise Lever::Error, "`Lever::Client#opportunities`'s new interface for returning an OpportunityCollection "\
+                              "does not allow for `id:` or `on_error:` keyword args"
+        end
+        query_params.merge!(contact_id: contact_id) unless contact_id.nil?
+
+        return Lever::OpportunityCollection.new(client: self, query_params: query_params)
+      end
```

Comment:
@hcliu can you read this over so we can double-check that this will not alter current functionality? I did some testing locally re: the use of the double-splat and all that, but another set of 👀  never hurts